src/renderer.c

```diff
@@ -9,60 +9,100 @@
 #include FT_OUTLINE_H
 #include FT_SYSTEM_H
 
-#ifdef _WIN32
-#include <windows.h>
-#include "utfconv.h"
-#endif
-
 #include "renderer.h"
 #include "renwindow.h"
 
-#define MAX_UNICODE 0x100000
-#define GLYPHSET_SIZE 256
-#define MAX_LOADABLE_GLYPHSETS (MAX_UNICODE / GLYPHSET_SIZE)
-#define SUBPIXEL_BITMAPS_CACHED 3
+// uncomment the line below for more debugging information through printf
+// #define RENDERER_DEBUG
 
 static RenWindow **window_list = NULL;
 static size_t window_count = 0;
-static FT_Library library;
 
 // draw_rect_surface is used as a 1x1 surface to simplify ren_draw_rect with blending
-static SDL_Surface *draw_rect_surface;
+static SDL_Surface *draw_rect_surface = NULL;
+static FT_Library library = NULL;
 
-static void* check_alloc(void *ptr) {
+#define check_alloc(P) _check_alloc(P, __FILE__, __LINE__)
+static void* _check_alloc(void *ptr, const char *const file, size_t ln) {
   if (!ptr) {
-    fprintf(stderr, "Fatal error: memory allocation failed\n");
+    fprintf(stderr, "%s:%zu: memory allocation failed\n", file, ln);
     exit(EXIT_FAILURE);
   }
   return ptr;
 }
 
 /************************* Fonts *************************/
 
+// approximate number of glyphs per atlas surface
+#define GLYPH_PER_ATLAS 256
```

Comment:
I'd use `GLYPHS` here.

Reply:
Not sure what you meant. You meant the macro name?

Reply:
Yeah using the plural.